src/Core/Services/Implementations/PolicyService.cs

```diff
@@ -127,5 +131,59 @@ public class PolicyService : IPolicyService
             await _policyRepository.UpsertAsync(policy);
             await _eventService.LogPolicyEventAsync(policy, Enums.EventType.Policy_Updated);
         }
+
+        public async Task<bool> PolicyAppliesToCurrentUserAsync(PolicyType policyType,
+            Func<Policy, bool> policyFilter)
+        {
+            if (policyFilter == null)
+            {
+                policyFilter = policy => true;
+            }
+
+            var userId = _currentContext.UserId.Value;
+            var policies = await _policyRepository.GetManyByUserIdAsync(userId);
+            var enabledPolicies = policies
+                .Where(p => p.Enabled && p.Type == policyType && policyFilter(p))
+                .Select(p => p.OrganizationId)
+                .ToHashSet();
+
+            if (!enabledPolicies.Any())
+            {
+                return false;
+            }
+
+            var orgUsers = await _organizationUserRepository.GetManyByUserAsync(userId);
+            
+            if (!orgUsers.Any())
+            {
+                return false;
+            }
+
+            var checkExemptionTasks = orgUsers
+                .Where(ou =>
+                    ou.Status >= OrganizationUserStatusType.Accepted &&
+                    enabledPolicies.Contains(ou.OrganizationId))
+                .Select(ou => _currentContext.ExemptFromPolicies(ou.OrganizationId))
+                .ToArray();
```

Comment:
This would also need to change from a call to _currentContext to an explicit check using the org user records you already pulled and targeting the userId that is passed in to the function.